docs/api.asciidoc

```diff
@@ -0,0 +1,159 @@
+ * `exc_info`: A `(type, value, traceback)` tuple as returned by https://docs.python.org/3/library/sys.html#sys.exc_info[`sys.exc_info()`]. If not provided, it will be captured automatically, if `capture_message()` was called in an `except` block.
+ * `date`: A `datetime.datetime` object representing the occurrence time of the error. If left empty, it defaults to `datetime.datetime.utcnow()`.
+ * `context`: A dictionary with contextual information. This dictionary must follow the 
+    https://www.elastic.co/guide/en/apm/server/current/error-api.html#error-context-schema[Context] schema definition.
+ * `custom`: A dictionary of custom data you want to attach to the event.
+
+Returns the id of the message as a string.
+
+[[transaction-api]]
+=== Transaction API
+
+[break]
+[[client-api-begin-transaction]]
+==== `Client.begin_transaction()`
+Begin tracking a transaction.
+Should be called e.g. at the beginning of a request or when starting a background task.
+
+[break]
+===== Arguments
+
+ * `transaction_type`: A string describing the type of the transaction, e.g. `'request'` or `'celery'`.
+
+[break]
+[[client-api-end-transaction]]
+==== `Client.end_transaction()`
+End tracking the transaction.
+Should be called e.g. at the end of a request or when ending a background task.
+
+[break]
+===== Arguments
+
+ * `name`: A string describing the name of the transaction, e.g. `process_order`.
+   This is typically the name of the view/controller that handles the request, or the route name.
+ * `result`: A string describing the result of the transaction.
+   This is typicaly the HTTP status code, or e.g. `'success'` for a background task.
+
+[break]
+[[api-elasticapm-instrument]]
+==== `elasticapm.instrument()`
+
+Instruments libraries automatically.
+This includes a wide range of standard library and 3rd party modules.
+A list of instrumented modules can be found in `elasticapm.intrumentation.register`.
+This function should be called as early as possibly in the startup of your application.
+For supported frameworks, this is called automatically.
+
+[break]
+[[api-set-transaction-name]]
+==== `elasticapm.set_transaction_name()`
+
+Override the name of the current transaction.
+For supported frameworks, the transaction name is determined automatically,
+and can be overridden using this function.
+
+===== Arguments
+
+ * `name`: A string describing name of the transaction
+
+[break]
+[[api-set-transaction-data]]
+==== `elasticapm.set_transaction_data()`
+
+Attach custom contextual data to current transaction.
+Supported frameworks will automatically attach information about the HTTP request and the logged in user.
+You can attach further data using this function.
+
+===== Arguments
+
+ * `data`: A dictionary with the data to be attached. This should be a flat key/value `dict` object.
+ * `_key`: The key to use for this `dict` object. Defaults to `custom`.
++
+NOTE: This should only be overridden by framework integrations.
```

Comment:
It would be great to add `optional` or `mandatory` to every parameter you describe. Sometimes you mention it is optional, sometimes you mention a default, sometimes none of both.

Reply:
For `begin_transaction` and `end_transaction` it is not very clear to me if the params are required or not.